.OpenAPIdocumentation.yaml

```diff
@@ -0,0 +1,98 @@
+openapi: 3.0.0
+info:
+  title: Ekadashi-server
+  version: 0.0.1
+servers:
+  - url: http://api.example.com/v1 #this field is required but inappropriate for current time
+paths:
+  /register:
+    post:
+      summary: Register a new user in ekadashi database.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Register'
+      responses:
+        '200':    # status code
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Register'
+        '400':    # status code
+          description: Cannot decode the request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IncorrectValue'
+        '401':    # status code
+          description: Incorrect username or password
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IncorrectValue'
+        '500':    # status code
+          description: Internal Server Error
+  /login:
+    post:
+      summary: Creates a new user.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Register'
+      responses:
+        '200':    # status code
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Register'
+        '400':    # status code
+          description: Cannot decode user
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IncorrectValue'
+        '401':    # status code
+          description: Incorrect username or password
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IncorrectValue'
+  /users:
+    get:
+      summary: Show all users in database
+      responses:
+        '200':    # status code
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: string
+            application/xml:
+              schema:
+                type: string
+        '500':   # status code
+          description: internal server error
+components:
+  schemas:
+    Register:
```

Comment:
better name it "registerRequest"

Reply:
also it is recommended to follow Go structure here, so you will have loginRequest and embed it here